src/Session/SessionAwareTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Pantheon\Terminus\Session;
+
+
+/**
+ * Provides the basic properties needed to fulfill the SessionAwareInterface.
+ * 
+ * Class SessionAwareTrait
+ * @package Pantheon\Terminus\Session
+ */
+trait SessionAwareTrait {
+
+  /**
+   * @var Session
+   */
+  protected $session;
+
+  /**
+   * @inheritdoc
+   */
+  public function setSession(Session $session)
+  {
+    $this->session = $session;
+
+    return $this;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function session()
+  {
+    if (!isset($this->session)) {
+      $this->setSession(new Session());
```

Comment:
Robo has some code like this that is an artifact of it moving from a lot of static code to DI. This does help move things along, but it can be kind of annoying, e.g. the recent Terminus bug "why isn't the output verbosity being set correctly?"  a: because Robo was kindly providing a new output object in absence of the right one.

So, you are right -- fine for now, best to move away from it when not needed anymore.